refactor (#334)
KxSystems/docs: docs/wp/data-visualization/index.md

dynamic and flexible. This will be explored below in
_Dynamic Parameters_.

#### Dynamic parameters

As mentioned above in _Simple parameters_, Tableau
parameters are limited to static values, and a single select option when

> placed in a view. However, there are a number of ways to make parameters

Reviewer: ```suggestion
placed in a view. However, there are several ways to make parameters
```

smarter, and can increase their usability and flexibility. Below, two
such methods are described. 

##### Predefining parameter options in a q function

From the previous example, the input parameter Category is limited to
single values. This can be made more flexible by
defining in the function a range of acceptable values.
In the example below, the
argument `` `all`` leads to a select with no restriction on `category`.

```q
func:{[mydate;mycategory]
  $[mycategory=`all;
    select from tab where date in mydate;
    select from tab where date in mydate, category in mycategory]
  };
```

Then `all` can be added to the list of predefined values in Tableau’s
definition of Category:

![](img/image11.png)

##### Parameters with calculated fields

Using parameters in conjunction with Tableau’s calculated-field
functionality can be a convenient and flexible tool in calculations as
well as graphical representation. This is useful when the output the
user wants to see is dependent on an input parameter, and a field needs
to be adjusted accordingly.

For example, in the user-defined `Calculation1` logic below, the quantity

> field will be divided by a different amount depending on the chosen

Reviewer: ```suggestion
field is divided by a different amount depending on the chosen
```

Category value. 

![](img/image12.png)

Below is sample output from when the user selects
a _Category_ value of `EQ`.

![](img/image13.png)

> In contrast, when the user selects `CORP` the calculated field will be

Reviewer: ```suggestion
In contrast, when the user selects `CORP` the calculated field is
```

This is not an ideal way to publish workbooks that are connected to a
kdb+ database, because connection details are stored
within the workbook itself. 

Take the following scenario:

-   A workbook has been developed in Tableau Desktop and is ready to share to the `Testing` partition in Tableau Server.
-   Throughout development, a development DSN has been used. But the workbook needs to be published to a UAT DSN.
-   So the DSN details need to be changed to the UAT DSN before publication.
-   The workbook again needs to be promoted, this time to the `Production` partition.
-   The workbook must be reopened, and the DSN details changed to the production DSN, before final promotion to `Production`.

This process is manual and prone to errors. 

### Publishing using tabcmd

For kdb+
connections, it is recommended to use the [`tabcmd`](https://help.tableau.com/current/server/en-us/tabcmd.htm) command-line utility

> which, among other things, enables the user to [publish to Tableau Server

Reviewer: ```suggestion
which, among other things, enables you to [publish to Tableau Server
```

> from the command line](https://help.tableau.com/current/server/en-us/tabcmd_cmd.htm#iddf805b62-18ff-4497-9245-adc6905b2084). This utility allows the user to deploy sheets

Reviewer: ```suggestion
from the command line](https://help.tableau.com/current/server/en-us/tabcmd_cmd.htm#iddf805b62-18ff-4497-9245-adc6905b2084). This utility allows you to deploy sheets
```
